modules/statistics/LocalStatsCollector.js

```diff
@@ -142,5 +143,45 @@ LocalStatsCollector.prototype.stop = function() {
  * @returns {boolean}
  */
 LocalStatsCollector.isLocalStatsSupported = function() {
-    return Boolean(context);
+    return Boolean(window.AudioContext);
 };
+
+/**
+ * Disconnects the audio context.
+ */
+LocalStatsCollector.disconnectAudioContext = async function() {
+    if (context) {
+        logger.info('Disconnecting audio context');
+        await context.close();
+        context = null;
+    }
+};
+
+/**
+ * Connects the audio context.
+ */
+LocalStatsCollector.connectAudioContext = function() {
+    if (!LocalStatsCollector.isLocalStatsSupported()) {
+        return;
+    }
+
+    logger.info('Connecting audio context');
+    context = new AudioContext();
+
+    // XXX Not all browsers define a suspend method on AudioContext. As the
+    // invocation is at the (ES6 module) global execution level, it breaks the
+    // loading of the lib-jitsi-meet library in such browsers and, consequently,
+    // the loading of the very Web app that uses the lib-jitsi-meet library. For
+    // example, Google Chrome 40 on Android does not define the method but we
+    // still want to be able to load the lib-jitsi-meet library there and
+    // display a page which notifies the user that the Web app is not supported
+    // there.
+    if (context.suspend) {
+        context.suspend();
```

Comment:
For symmetry, and to avoid surprises, we should `await` here. Also, AFAICT suspend has been supported long enough that we don't need this check. See: https://caniuse.com/mdn-api_audiocontext_suspend

Reply:
I had it with await initially, but it seem to never resolve the promise for some reason  as the for after connectAudioContext() call was not executing

Reply:
Can you check if there any reported bugs? If so, please put a comment.

Reply:
Ok so the AudioContext is in `suspended` state right after the instance is created, that probably explains why the await context.suspend() did not resolve. For testing I added a state change listener on the context, and at some point it did go to `running` state (probably when `oncomplete` event triggers)
 
So there are 2 choices here:
1. Call `await context.suspend()` when state first changes to `running`  ( or on the `oncomplete` event handler- have to check if that event can be used)
2. Leave it as it is without awaiting- as from my test it does do suspend - as the context does not get to `running` state anymore -  and add the `JitsiTrackEvents.NO_DATA_FROM_SOURCE` listener only for safari ios - this will basically keep the current logic for all other browser, as existing code does not await for suspend  call either.


@saghul  WDYT?

Reply:
LEt's go with option 2 then.

Reply:
I did go with option 2 in the end, as I was not able to make option 1 work no matter what I tried. Since I added the condition to add the listener for audio muted only on iOS browsers, the behavior for the rest of OS's is just as before.